--- src/domains/afternomie/afternomie-view.svelte
+++ src/domains/afternomie/afternomie-view.svelte
@@ -212,12 +212,14 @@
    }

    function key(length){
     var chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
    var charLength = chars.length;
    var result = '';
+   var randomValues = new Uint32Array(length);
+   globalThis.crypto.getRandomValues(randomValues);
    for ( var i = 0; i < length; i++ ) {
-      result += chars.charAt(Math.floor(Math.random() * charLength));
+      result += chars.charAt(randomValues[i] % charLength);
    }
    return result;
    }
